kernel_crawler/__init__.py

```diff
@@ -44,16 +47,25 @@ def to_driverkit_config(d, res):
 
     return dk_configs
 
-def crawl_kernels(distro, version, arch, to_driverkit):
+def crawl_kernels(distro, version, arch, image, to_driverkit):
     ret = {}
 
     for distname, dist in DISTROS.items():
         if distname == distro or distro == "*":
-            d = dist(arch)
-            res = d.get_package_tree(version)
-            if to_driverkit:
-                ret[distname] = to_driverkit_config(d, res)
+            if issubclass(dist, repo.ContainerDistro):
+                if image:
```

Comment:
Here i'd expect a loop on images; am i missing anything obvious here?

Reply:
There is only one container distro (Redhat) so all of the `images` are passed to that. `RedhatContainer` will take care of looping the images so that there is only one target returned (Redhat) but a variety of kernels based off of the images provided. This matches the behaviour for the mirrors where mutliple versions of a repo are checked but returned under the one target name.

Reply:
I see, the loop is implemented in `get_kernel_versions`!
I'd make it explicit here instead; that would be helpful if/when adding new container distros: the new $DistroContainer would not have to care about the loop.

I think we are able to still return a list for a single distro in any case, right?

Reply:
No problem sir!
And sorry for bothering you, but i think it is better to keep it explicit so that `get_kernel_versions` only works on a single image :)

Reply:
Image loop moved from the ```get_kernel_versions()``` function as requested 👍